routes/web/project/user.rb

```diff
@@ -6,16 +6,25 @@ class CloverWeb
 
     r.get true do
       @users = Serializers::Account.serialize(@project.accounts)
+      @invitations = Serializers::ProjectInvitation.serialize(@project.invitations)
 
       view "project/user"
     end
 
     r.post true do
       email = r.params["email"]
-      # Don't invite deleted accounts
-      user = Account.exclude(status_id: 3)[email: email]
 
-      user&.associate_with_project(@project)
+      if (user = Account.exclude(status_id: 3)[email: email])
+        user.associate_with_project(@project)
+      elsif ProjectInvitation[project_id: @project.id, email: email]
+        flash["error"] = "'#{email}' already invited to join the project."
+        r.redirect "#{@project.path}/user"
+      elsif @project.invitations.count >= 50
```

Comment:
```suggestion
      elsif @project.invitations_dataset.count >= 50
```

Perhaps? No need to load the tuples.

Reply:
Makes sense. Done. I was expecting this comment from you 😄